libtiledbsoma/src/soma_reader.cc

```diff
@@ -97,6 +96,18 @@ SOMAReader::SOMAReader(
     if (!column_names.empty()) {
         mq_->select_columns(column_names);
     }
+
+    if (result_order != "auto") {  // default "auto" is set in soma_reader.h
```

Comment:
@gspowley @johnkerl style question...on the one hand, in pure C/C++ usually one tries to use an enum or something instead of comparing string constants. On the other hand, I can see that taking `result_order` (and eventually `batch_size`) as strings is convenient for calling in from Python & R though. Is that the winning "hand"?

Reply:
LGTM, I like the convenience of comparing string constants in this case.

Reply:
+1